airbyte-integrations/connectors/source-datadog/source_datadog/manifest.yaml

```diff
@@ -1,276 +1,2068 @@
-    $parameters:
-      name: "incidents"
-      primary_key: "id"
-      path: "incidents"
-    retriever:
-      $ref: "#/definitions/retriever_v2"
-      paginator:
-        $ref: "#/definitions/base_paginator"
+      type: ApiKeyAuthenticator
+      api_token: '{{ config["api_key"] }}'
+      inject_into:
+        type: RequestOption
+        field_name: DD-API-KEY
+        inject_into: header
+  base_requester_v2:
+    type: HttpRequester
+    url_base: https://api.{{ config['site'] }}/api/v2/
+    authenticator:
+      type: ApiKeyAuthenticator
+      api_token: '{{ config["api_key"] }}'
+      inject_into:
+        type: RequestOption
+        field_name: DD-API-KEY
+        inject_into: header
 
-  incident_teams_stream:
-    $ref: "#/definitions/base_stream"
-    $parameters:
-      name: "incident_teams"
-      primary_key: "id"
-      path: "teams"
-    retriever:
-      $ref: "#/definitions/retriever_v2"
-      paginator:
-        $ref: "#/definitions/base_paginator"
+streams:
+  - $ref: "#/definitions/streams/audit_logs"
+  - $ref: "#/definitions/streams/dashboards"
+  - $ref: "#/definitions/streams/downtimes"
+  - $ref: "#/definitions/streams/incident_teams"
+  - $ref: "#/definitions/streams/incidents"
+  - $ref: "#/definitions/streams/logs"
+  - $ref: "#/definitions/streams/metrics"
+  - $ref: "#/definitions/streams/monitors"
+  - $ref: "#/definitions/streams/service_level_objectives"
+  - $ref: "#/definitions/streams/synthetic_tests"
+  - $ref: "#/definitions/streams/users"
 
-  users_stream:
-    $ref: "#/definitions/base_stream"
-    $parameters:
-      name: "users"
-      primary_key: "id"
-      path: "users"
-    retriever:
-      $ref: "#/definitions/retriever_v2"
-      paginator:
-        $ref: "#/definitions/base_page_paginator"
+spec:
+  type: Spec
+  connection_specification:
+    type: object
+    $schema: http://json-schema.org/draft-07/schema#
+    required:
+      - api_key
+      - application_key
+    properties:
+      api_key:
+        type: string
+        description: Datadog API key
+        order: 0
+        title: API Key
+        airbyte_secret: true
+      application_key:
+        type: string
+        description: Datadog application key
+        order: 1
+        title: Application Key
+        airbyte_secret: true
+      query:
+        type: string
+        description: >-
+          The search query. This just applies to Incremental syncs. If empty,
+          it'll collect all logs.
+        order: 2
+        title: Query
+      start_date:
+        type: string
+        description: >-
+          UTC date and time in the format 2017-01-25T00:00:00Z. Any data before
+          this date will not be replicated. This just applies to Incremental
+          syncs.
+        order: 3
+        title: Start date
+        pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$
+        examples:
+          - "2022-10-01T00:00:00Z"
+        default: "2024-12-01T00:00:00Z"
```

Comment:
Hmmm, that would mean that the start date by default is in the future. 

What if instead of setting the default, we make this field optional, and use a srtring interpolation to default to 1 year ago, but dynamically? 

Similar to this: https://github.com/airbytehq/airbyte/blob/526bdee712498e04d97d3cabe1dbc7cb4d82a19a/airbyte-integrations/connectors/source-linkedin-pages/manifest.yaml#L649

@girarda what's a better practice?

Reply:
Changed my mind: let's perhaps default to a specific start date, but in the past — smaller change. 2024-01-01 maybe?